lib/rsa.js

```diff
@@ -887,6 +889,30 @@ pki.rsa.generateKeyPair = function(bits, e, options, callback) {
   // key generation code if available and if parameters are acceptable
   if(!forge.options.usePureJavaScript && callback &&
     bits >= 256 && bits <= 16384 && (e === 0x10001 || e === 3)) {
+    if(forge.util.isNodejs && _detectNodeCrypto('generateKeyPair')) {
```

Comment:
I wonder if this conditional might be simplified.  if the `util.isNodejs` call is necessary, maybe it could go into `detectNodeCrypto`.

Reply:
Yep. Will do.